### PR TITLE
add tests for serve methods

### PR DESCRIPTION
**Pull request by willnorris**
Also check error returned from tmpl.Execute. Refactor currentUser to make the logic a little simpler, and make it a package var for easier testing.

This doesn't make any behavioral changes (aside from checking the tmpl.Execute error).  Instead, it's focused on making the  next pass at #72 much simpler to review.

### CODE REVIEW

File: golink_test.go [hunk 1]
```diff
@@ -4,17 +4,162 @@
...
+			currentUser: func(*http.Request) (string, error) { return "", nil },
+			wantStatus:  http.StatusFound,
+			wantLink:    "http://who/",
+		},
+		{
+			name:       "user link",
+			link:       "/me",
+			wantStatus: http.StatusFound,
+			wantLink:   "/who/foo@example.com",
+		},
```
Comment (shayne) [inline golink_test.go:55]: Does it make sense to add any negative tests here?

File: golink_test.go [hunk 2]
```diff
@@ -85,8 +235,8 @@ func TestExpandLink(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := expandLink(tt.long, expandEnv{Now: tt.now, Path: tt.remainder, User: tt.user})
-			if err != nil {
-				t.Fatalf("expandLink(%q): %v", tt.long, err)
+			if (err != nil) != tt.wantErr {
```
Comment (shayne) [inline golink_test.go:238]: 😵‍💫

**Review by shayne — APPROVED** [review]
Just a couple of comments. Not totally in love with `currentUser` as the name for the pkg level func, but I don't have a better idea.

File: golink_test.go (hunk 1 above)
Comment (willnorris) [reply golink_test.go:56]: done

**Pull request merged**